src/canopy/chat_engine/query_generator/last_message.py

```diff
@@ -0,0 +1,28 @@
+from typing import List
+
+from canopy.chat_engine.query_generator import QueryGenerator
+from canopy.models.data_models import Messages, Query
+
+
+class LastMessageQueryGenerator(QueryGenerator):
+    """
+        Returns the last message as a query without running any LLMs. This can be
+        considered as the most basic query generation. Please use other query generators
+        for more accurate results.
+    """
+    def generate(self,
+                 messages: Messages,
+                 max_prompt_tokens: int) -> List[Query]:
+        """
+            max_prompt_token is dismissed since we do not consume any token for
+            generating the queries.
+        """
+        if len(messages) == 0:
+            raise ValueError("Passed chat history does not contain any messages."
+                             " Please include at least one message in the history.")
+        return [Query(text=messages[-1].content)]
```

Comment:
Please take the last message which is a `UserMessage` (`message.role == 'Role.user'` or `isinstance()`).  
If you go backwards on the messages and can't find a `UserMessage` - then raise an error

Reply:
This sounds a bit weird to me. It's more likely that if the last message is not user message, there is something wrong in the pipeline or usage of this class, so IMO it's better to raise an error here instead of silently do something that might not be expected by the user

Reply:
I see your point @igiloh-pinecone but agree with @acatav. Maybe I can raise if the history is empty or the last message is not a user message @igiloh-pinecone wdyt?